### PR TITLE
Available Interface

### PR DESCRIPTION
**Pull request by walkohm**
Swap isAvailable func to Available struct with methods IsAvailable and String and pass entity id to DoAction func.

### CODE REVIEW

File: multistate.go [hunk 1]
```diff
@@ -216,7 +216,7 @@ func (m *Multistate) GetStateActions(ctx context.Context, state uint64) []string
 	return nil
 }
 
-func (m *Multistate) DoAction(ctx context.Context, entity Entity, action string, opts ...interface{}) (uint64, error) {
+func (m *Multistate) DoAction(ctx context.Context, entity Entity, id interface{}, action string, opts ...interface{}) (uint64, error) {
```
Comment (sergei-svistunov) [inline multistate.go:219]: `id` is already in entity, you don't need a separate field.

Comment (walkohm) [reply multistate.go:219]: But we can't  get it

File: example_fromstruct_test.go [hunk 2]
```diff
@@ -52,12 +52,13 @@ func (*ExampleEntity) StartAction(ctx context.Context) (context.Context, error)
...
 func (*ExampleEntity) SetState(context.Context, uint64, ...interface{}) error   { return nil }
 func (*ExampleEntity) EndAction(context.Context, error) error                   { return nil }
+func (e *ExampleEntity) GetId() interface{}                                     { return e.Id }
 
 func ExampleNewFromStruct() {
 	mst := multistate.NewFromStruct(&ExampleImpl{})
 
 	e := &ExampleEntity{Id: 100}
-	_, err := mst.DoAction(context.Background(), e, "test")
+	_, err := mst.DoAction(context.Background(), e, 1, "test")
```
Comment (sergei-svistunov) [inline example_fromstruct_test.go]: Unnecessary argument  `1`

Comment (walkohm) [reply example_fromstruct_test.go]: I removed it.

**Pull request merged**